### PR TITLE
James 1744 2

### PR DESCRIPTION
**Pull request by Laura-Royet**

### CODE REVIEW

**Comment by rouazana** [comment]
:+1:

**Comment by mbaechler** [comment]
:+1:

**Comment by chibenwa** [comment]
Can't this depend on the newly introduced "HtmlTextExtractor" ?

File: server/protocols/jmap/src/test/java/org/apache/james/jmap/methods/GetMessagesMethodTest.java [hunk 1]
```diff
@@ -37,8 +37,12 @@
 import org.apache.james.jmap.model.GetMessagesRequest;
 import org.apache.james.jmap.model.GetMessagesResponse;
 import org.apache.james.jmap.model.Message;
+import org.apache.james.jmap.model.MessageFactory;
```
Comment (chibenwa) [inline server/protocols/jmap/src/test/java/org/apache/james/jmap/methods/GetMessagesMethodTest.java]: I do not find your MessageFactory class in this diff...

**Comment by mbaechler** [comment]
:+1:

**Comment by chibenwa** [comment]
:+1:

File: server/container/guice/guice-common/src/main/java/org/apache/james/jmap/JMAPCommonModule.java [hunk 2]
```diff
@@ -54,6 +55,7 @@ protected void configure() {
         bind(MailSpool.class).in(Scopes.SINGLETON);
         bind(MailFactory.class).in(Scopes.SINGLETON);
         bind(AutomaticallySentMailDetectorImpl.class).in(Scopes.SINGLETON);
+        bind(MessageFactory.class).in(Scopes.SINGLETON);
```
Comment (aduprat) [inline server/container/guice/guice-common/src/main/java/org/apache/james/jmap/JMAPCommonModule.java]: You may also add `MessagePreviewGenerator`

File: server/protocols/jmap/pom.xml [hunk 3]
```diff
@@ -334,14 +311,32 @@
...
+                </dependency>
+                <dependency>
                     <groupId>org.eclipse.jetty</groupId>
                     <artifactId>jetty-server</artifactId>
                     <scope>test</scope>
                 </dependency>
                 <dependency>
-                    <groupId>org.eclipse.jetty</groupId>
-                    <artifactId>jetty-servlet</artifactId>
-                    <scope>test</scope>
```
Comment (rouazana) [inline server/protocols/jmap/pom.xml]: Why are you removing this one?

Comment (Laura-Royet) [inline server/protocols/jmap/pom.xml]: It's a mistake :(

File: server/protocols/jmap/src/test/java/org/apache/james/jmap/model/MailboxMessageTest.java [hunk 4]
```diff
@@ -241,8 +255,7 @@ public void emptyMailShouldBeLoadedIntoMessage() throws Exception {
                 new PropertyBuilder(),
                 MAILBOX_ID);
         testMail.setModSeq(MOD_SEQ);
-        
-        Message testee = Message.fromMailboxMessage(testMail, ImmutableList.of(), x -> MessageId.of("user|box|" + x));
+        Message testee = new MessageFactory(messagePreview).fromMailboxMessage(testMail, ImmutableList.of(), x -> MessageId.of("user|box|" + x));
```
Comment (rouazana) [inline server/protocols/jmap/src/test/java/org/apache/james/jmap/model/MailboxMessageTest.java]: You do you use new MessageFactory while you already defined one in the setup?

File: server/protocols/jmap/src/test/java/org/apache/james/jmap/model/MessagePreviewGeneratorTest.java [hunk 5]
```diff
@@ -34,21 +47,30 @@ public void forHTMLBodyShouldReturnTruncatedStringWithoutHtmlTagWhenStringContai
...
 
     @Test
     public void forHTMLBodyShouldReturnStringContainingEmptyWhenEmptyString() {
         //Given
+        String body = "" ;
         String expected = "(Empty)" ;
         //When
-        String actual = MessagePreview.forHTMLBody(Optional.empty());
+        when(htmlTextExtractor.toPlainText(body))
+        .thenReturn(expected);
```
Comment (rouazana) [inline server/protocols/jmap/src/test/java/org/apache/james/jmap/model/MessagePreviewGeneratorTest.java]: indent

File: server/protocols/jmap/src/test/java/org/apache/james/jmap/model/MessagePreviewGeneratorTest.java [hunk 6]
```diff
@@ -85,14 +107,16 @@ public void asTextShouldReturnStringWithoutHtmlTag() {
         String body = "This is a <b>HTML</b> mail";
         String expected = "This is a HTML mail";
         //When
-        String actual = MessagePreview.asText(body);
+        when(htmlTextExtractor.toPlainText(body))
+        .thenReturn(expected);
```
Comment (rouazana) [inline server/protocols/jmap/src/test/java/org/apache/james/jmap/model/MessagePreviewGeneratorTest.java]: indent

File: server/protocols/jmap/src/test/java/org/apache/james/jmap/model/MessagePreviewGeneratorTest.java [hunk 7]
```diff
@@ -101,7 +125,9 @@ public void asTextShouldReturnEmptyStringWhenEmptyString() {
         String body = "";
         String expected = "";
         //When
-        String actual = MessagePreview.asText(body);
+        when(htmlTextExtractor.toPlainText(body))
+        .thenReturn(expected);
```
Comment (rouazana) [inline server/protocols/jmap/src/test/java/org/apache/james/jmap/model/MessagePreviewGeneratorTest.java]: indent

File: server/protocols/jmap/pom.xml [hunk 8]
```diff
@@ -234,6 +234,11 @@
                     <artifactId>james-server-util-java8</artifactId>
                 </dependency>
                 <dependency>
+                    <groupId>org.apache.commons</groupId>
+                    <artifactId>commons-lang3</artifactId>
+                    <version>3.3.2</version>
+                </dependency>
+                <dependency>
```
Comment (rouazana) [inline server/protocols/jmap/pom.xml]: This fixup should not be a fixup of sorting of dependencies, but a fixup of the commit when you create the MessagePreview(Generator)

File: server/protocols/jmap/src/test/java/org/apache/james/jmap/model/MailboxMessageTest.java (hunk 4 above)
Comment (Laura-Royet) [inline server/protocols/jmap/src/test/java/org/apache/james/jmap/model/MailboxMessageTest.java]: Ok. I remove the useless second instanciation.

File: server/protocols/jmap/src/test/java/org/apache/james/jmap/model/MessagePreviewGeneratorTest.java (hunk 5 above)
Comment (Laura-Royet) [inline server/protocols/jmap/src/test/java/org/apache/james/jmap/model/MessagePreviewGeneratorTest.java]: Ok. Done.

File: server/protocols/jmap/src/test/java/org/apache/james/jmap/model/MessagePreviewGeneratorTest.java (hunk 6 above)
Comment (Laura-Royet) [inline server/protocols/jmap/src/test/java/org/apache/james/jmap/model/MessagePreviewGeneratorTest.java]: Ok. Done

**Comment by Laura-Royet** [comment]
test this please

File: server/protocols/jmap/src/test/java/org/apache/james/jmap/methods/GetMessagesMethodTest.java (hunk 1 above)
Comment (Laura-Royet) [inline server/protocols/jmap/src/test/java/org/apache/james/jmap/methods/GetMessagesMethodTest.java]: Ok. Class added.

File: server/protocols/jmap/pom.xml (hunk 8 above)
Comment (Laura-Royet) [inline server/protocols/jmap/pom.xml]: oK. Done.

File: server/protocols/jmap/pom.xml [hunk 9]
```diff
@@ -279,6 +284,21 @@
...
+                    <groupId>org.jgrapht</groupId>
+                    <artifactId>jgrapht-core</artifactId>
+                    <version>0.9.1</version>
+                </dependency>
+                <dependency>
+                    <groupId>org.jsoup</groupId>
+                    <artifactId>jsoup</artifactId>
+                    <version>1.9.2</version>
+                </dependency>
+                <dependency>
```
Comment (rouazana) [inline server/protocols/jmap/pom.xml]: Sorry, but your Maven imports are still no correct:
- you're moving 2 imports while it should be done in the other commit (+test modifier added)
- more important, you are adding here jsoup while you don't need it at all

File: server/protocols/jmap/pom.xml [hunk 10]
```diff
@@ -333,22 +305,45 @@
...
                     <groupId>org.eclipse.jetty</groupId>
-                    <artifactId>jetty-server</artifactId>
+                    <artifactId>jetty-servlet</artifactId>
                     <scope>test</scope>
                 </dependency>
                 <dependency>
                     <groupId>org.eclipse.jetty</groupId>
-                    <artifactId>jetty-servlet</artifactId>
+                    <artifactId>jetty-server</artifactId>
```
Comment (rouazana) [inline server/protocols/jmap/pom.xml]: are you sure jetty-servlet should be before jetty-server?

File: server/protocols/jmap/pom.xml (hunk 9 above)
Comment (Laura-Royet) [inline server/protocols/jmap/pom.xml]: done.

File: server/protocols/jmap/pom.xml (hunk 10 above)
Comment (Laura-Royet) [inline server/protocols/jmap/pom.xml]: note

**Comment by Laura-Royet** [comment]
https://github.com/linagora/james-project/pull/280

**Pull request closed without merging**